hpobench/config.py

```diff
@@ -144,15 +162,57 @@ def __parse_config(self):
         self.pyro_connect_max_wait = int(read_config.get('pyro_connect_max_wait',
                                                          self.pyro_connect_max_wait))
 
-    def _check_version(self):
+    @staticmethod
+    def _check_version(config_version, hpobench_version):
         """ Check if the version of the configuration file matches the hpobench version.
             Ignore the `dev` tag at the end.
+
+            It may happen that the configuration file changes with a new version of the hpobench. In this case,
+            we want to raise a warning to show the user that there could be potential error.
+
+            But since the configuration file does not change every new hpobench version, we group versions together that
+            have similar configuration files. If the current version is equal to a version from the same partition, then
+            no problem can occur.
```

Comment:
Is this true? If we have, e.g. 2 valid configuration ranges X and Y. Is there a problem if a user uses HPOBench X, a config file written by X, but the containers are built with version Y?

Reply:
yes, this can cause a problem. Maybe you remember that one of the first versions of the hpobench has parsed the configuration file with the configparser-module. However, the current versions try to parse it as a yaml file. This constellation would crash badly. 

The point is that also the container needs to read the configuration file. For this, we need to ensure that the configuration file "reading mechanism" of the container is not very different from the used repo.